lib/ansible/plugins/filter/core.py

```diff
@@ -265,7 +265,23 @@ def get_encrypted_password(password, hashtype='sha512', salt=None, salt_size=Non
 
     hashtype = passlib_mapping.get(hashtype, hashtype)
     try:
-        return passlib_or_crypt(password, hashtype, salt=salt, salt_size=salt_size, rounds=rounds, ident=ident)
+        if encrypt.PASSLIB_AVAILABLE:
+            if hashtype == 'bcrypt':
+                if not rounds:
+                    rounds = 12
+                if not salt:
+                    salt = encrypt.random_salt(length=22)
+                if not ident:
+                    ident = "2b"
+            if hashtype == 'sha256_crypt' or hashtype == 'sha512_crypt':
+                if hashtype == 'sha256_crypt' and not rounds:
+                    rounds = 535000
+                if hashtype == 'sha512_crypt' and not rounds:
+                    rounds = 656000
+                if not salt:
+                    salt = encrypt.random_salt(length=16)
+
```

Comment:
The default values are already been handled [here](https://github.com/ansible/ansible/blob/afe6eb574ef8e53e805cc1ff2b25dd9108b29e1c/lib/ansible/utils/encrypt.py#L79)